lib/saito/peer.js

```diff
@@ -214,81 +215,84 @@ console.log("done emitting handshake complete");
+        console.log("X: " + reconstructed_message);
+        console.log("Y: " + JSON.stringify(reconstructed_data));
 
-	let mycallback = function(response_object) {
-          //peer.sendResponseFromStr(msg.message_id, "Response A");
-          //peer.sendResponseFromStr(msg.message_id, "Response A");
-	}
+        let mycallback = function (response_object) {
+            peer.sendResponse(msg.message_id, response_object);
+        }
 
         switch (msg.message_name) {
-          case 'block':
-console.log("old block request");
-            //this.handleBlockRequest(message);
-            break;
-          case 'missing block':
-console.log("old missing block request");
-            //this.handleMissingBlockRequest(message, mycallback);
-            break;
-          case 'blockchain':
-console.log("old blockchain request");
-            //this.handleBlockchainRequest(message);
-            break;
-          case 'transaction':
-console.log("old transaction request");
-            //this.handleTransactionRequest(message, mycallback);
-            break;
-          case 'keylist':
-console.log("old keyist request");
-            //this.handleKeylistRequest(message);
-            break;
-          default:
-
-
-            //
-            // if the attached data is a transaction, ensure MSG field exists
-            //
-            // transaction.msg included for backwards compatibility
-            //
-            if (reconstructed_data) {
-              if (reconstructed_data.transaction) {
-                if (reconstructed_data.transaction.m) {
-                  // backwards compatible - in case modules try the old fashioned way
-                  message.data.transaction.msg = JSON.parse(this.app.crypto.base64ToString(message.data.transaction.m));
-                  message.data.msg = message.data.transaction.msg;
+            case 'block':
+                console.log("old block request");
+                //this.handleBlockRequest(message);
+                break;
+            case 'missing block':
+                console.log("old missing block request");
+                //this.handleMissingBlockRequest(message, mycallback);
+                break;
+            case 'blockchain':
+                console.log("old blockchain request");
+                //this.handleBlockchainRequest(message);
+                break;
+            case 'transaction':
+                console.log("old transaction request");
+                //this.handleTransactionRequest(message, mycallback);
+                break;
+            case 'keylist':
+                console.log("old keyist request");
+                //this.handleKeylistRequest(message);
+                break;
+            default:
+
+
+                //
+                // if the attached data is a transaction, ensure MSG field exists
+                //
+                // transaction.msg included for backwards compatibility
+                //
+                if (reconstructed_data) {
```

Comment:
Shouldn't [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) work here?:

`if (reconstructed_data?.transaction?.m) {`

Reply:
only formatted the current code here.